views/js/previewer/plugins/tools/scale/scale.js

```diff
@@ -0,0 +1,521 @@
+               self.api.onDeviceTypeChange(event);
+               self.trigger('preview-scale-device-type', event.target.value);
+
+            });
+
+            /**
+             *  when user changes mobile device model he/she want to test item in
+             *  @event scale#preview-scale-device-mobile-type
+             */
+            this.controls.$mobileDevices.children('.mobile-device-selector').on('change', function (event) {
+                self.api.onDeviceChange(event);
+                self.trigger('preview-scale-device-mobile-type', event.target.value);
+
+            });
+
+            /**
+             *  when user changes mobile device screen orientation user want to test item in
+             *  @event scale#preview-scale-device-mobile-orientation-type
+             */
+            this.controls.$mobileDevices.children('.mobile-orientation-selector').on('change', function (event) {
+                self.api.onOrientationChange(event);
+                self.trigger('preview-scale-device-mobile-orientation-type', event.target.value);
+            });
+
+            /**
+             *  when user changes mobile device model he/she want to test item in
+             *  @event scale#preview-scale-device-desktop-type
+             */
+            this.controls.$desktopDevices.on('change', function (event) {
+                self.api.onDeviceChange(event);
+                self.trigger('preview-scale-device-desktop-type', event.target.value);
+            });
+
+            /**
+             * adjust device frame position and size when browser size change
+             */
+            $(window).on('resize orientationchange', function () {
+                if(self.controls.$scaleWrapper){
+                    this.api.setupScreenSize();
+                    this.api.updateStandardPreviewSize();
+                    this.api.scaleFrame();
+                }
+            });
+
+            this.disable();
+
+            testRunner
+                .on('render', function () {
+                    if (self.api.isPluginAllowed()) {
+                        self.api.composeControlsByDeviceType(DEFAULT_TYPE);
+                    } else {
+                        self.hide();
+                    }
+                })
+                .on('enablenav', function () {
+                    self.enable();
+                })
+                .on('disablenav', function () {
+                    self.disable();
+                });
+        },
+
+        /**
+         * Called during the runner's render phase
+         * Rendeds plugins controlls on proper place
+         */
+        render: function render() {
+            var $headerControls = this.getAreaBroker().getHeaderArea();
+
+            $headerControls.append(this.controls.$deviceTypes);
+            $headerControls.append(this.controls.$mobileDevices);
+            $headerControls.append(this.controls.$desktopDevices);
+
+            selecter($headerControls);
+
```

Comment:
There are inconsistencies in the last lines of your functions, sometimes a blank line, sometimes not. Could you make them consistent ?